filer/migrations/0012_file_mime_type.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 2.1.9 on 2019-06-25 19:12
+
+import mimetypes
+
+from django.db import migrations, models
+
+import filer.models.filemodels
+
+
+def guess_mimetypes(apps, schema_editor):
+    FileModel = apps.get_model('filer', 'File')
+    for fileobj in FileModel.objects.all():
+        fileobj.mime_type, _ = mimetypes.guess_type(fileobj.file.path)
```

Comment:
Just telling from the top of my head, I need to check this better, but this *might* not work on all the storage engines

Reply:
> mimetypes.guess_type(url, strict=True)
Guess the type of a file based on its filename or URL, given by url. The return value is a tuple (type, encoding) where type is None if the type can’t be guessed (missing or unknown suffix) or a string of the form 'type/subtype', usable for a MIME content-type header.

According to the Python docs only the file name is used to guess the mime-type, rather than opening the file and using a magic header to determine the type. Therefore this should also work on S3BOTO.

Reply:
Apparently not

``` python
>>> import mimetypes
>>> from filer.models import File
>>> file_obj = File.objects.first()
>>> print(file_obj.file.path)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/ubuntu/virtualenvs/project/lib/python3.7/site-packages/django/db/models/fields/files.py", line 57, in path
    return self.storage.path(self.name)
  File "/home/ubuntu/virtualenvs/project/lib/python3.7/site-packages/django/core/files/storage.py", line 109, in path
    raise NotImplementedError("This backend doesn't support absolute paths.")
NotImplementedError: This backend doesn't support absolute paths.
>>> mimetypes.guess_type(file_obj.file.path)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/ubuntu/virtualenvs/project/lib/python3.7/site-packages/django/db/models/fields/files.py", line 57, in path
    return self.storage.path(self.name)
  File "/home/ubuntu/virtualenvs/project/lib/python3.7/site-packages/django/core/files/storage.py", line 109, in path
    raise NotImplementedError("This backend doesn't support absolute paths.")
NotImplementedError: This backend doesn't support absolute paths.
```

Reply:
but `url` will do:

``` python
>>> mimetypes.guess_type(file_obj.file.url)
('image/png', None)
```